backend/src/packages/users/user.repository.ts

```diff
@@ -163,12 +163,24 @@ class UserRepository implements IRepository {
   }
 
   public async getAllAuthors(): Promise<UserDetailsModel[] | null> {
-    const authors = await UserDetailsModel.query().distinctOn('userId')
-      .whereRaw(`
-      EXISTS(SELECT 1
-      FROM articles,user_details
-      WHERE articles.user_id = user_details.user_id)
-    `);
+    const authors = await UserDetailsModel.query()
+      .from((builder) => {
+        void builder
+          .distinctOn('userId')
+          .select('userId', 'firstName', 'lastName')
+          .from('user_details')
+          .whereRaw(
+            `
+        EXISTS(SELECT 1
+        FROM articles
+        WHERE articles.user_id = user_details.user_id)
```

Comment:
I noticed that all users are displayed as authors, even if no articles are written by the user. Updated query to resolve this.